statefun-sdk-js/src/storage.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import "./generated/request-reply_pb";
+import {TypedValueSupport} from "./types";
+import {Type, ValueSpec} from "./core";
+
+const M = global.proto.io.statefun.sdk.reqreply.FromFunction.PersistedValueMutation;
+
+const DEL = global.proto.io.statefun.sdk.reqreply.FromFunction.PersistedValueMutation.MutationType['DELETE'];
+
+// noinspection JSUnresolvedVariable
+const MOD = global.proto.io.statefun.sdk.reqreply.FromFunction.PersistedValueMutation.MutationType['MODIFY'];
+
+// noinspection JSValidateJSDoc
+class Value {
+    readonly #name;
```

Comment:
These are all untyped

Reply:
I thought that the types are inferred from the construction, good to know that this is not the case.
I'm fixing it.